### PR TITLE
Send ext.app.userId to One Collector for the default target

### PR DESCRIPTION
**Pull request by guperrot**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

* Send ext.app.userId to One Collector for the default target.
* Handle validation.

### CODE REVIEW

File: sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTest.java [hunk 1]
```diff
@@ -367,6 +367,45 @@ public void trackEventWithInvalidFlags() {
         AppCenterLog.warn(eq(AppCenter.LOG_TAG), anyString());
     }
 
+    @Test
+    public void trackEventWithUserIdWhenConfiguredForTarget() {
+        when(mAppCenter.getUserId()).thenReturn("alice");
```
Comment (jaelim-ms) [inline sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTest.java]: nit:Can you use prefix for the userId when mocking the getter? Just to make things clear, nothing more than that.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/models/one/AppExtension.java [hunk 2]
```diff
@@ -32,6 +32,11 @@
      */
     private static final String LOCALE = "locale";
 
+    /**
+     * User ID.
+     */
+    private static final String USER_ID = "userId";
```
[the remark this answers is not in the document]
Comment (guperrot) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/models/one/AppExtension.java]: This is the correct order in Bond we tried to align in Android.

File: sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTest.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTest.java]: ok

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/models/one/AppExtension.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaelim-ms) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/models/one/AppExtension.java]: I got confused. I thought this one is for UserExtension. Never mind.

**Pull request merged**